window/src/os/wayland/seat.rs

```diff
@@ -65,9 +65,11 @@ impl SeatHandler for WaylandState {
         _conn: &Connection,
         _qh: &QueueHandle<Self>,
         _seat: WlSeat,
-        _capability: smithay_client_toolkit::seat::Capability,
+        capability: smithay_client_toolkit::seat::Capability,
     ) {
-        todo!()
+        if capability == Capability::Keyboard && self.keyboard.is_some() {
+            self.keyboard = None;
+        }
```

Comment:
Should we keep the TODO in there for when the capability is not a keyboard? Or do we only ever deal with keyboard capabilities?

e.g.

```
if capability == Capability::Keyboard {
    if self.keyboard.is_some() {
        self.keyboard = None;
    }
} else {
    todo!();
}
```

Reply:
I don't know if there are more, according to the wayland spec you should use this hook to destroy and resources, from what I can tell, this was the only one we set that's why I added the removal as well. I don't think we should keep the todo! since that actually will crash the app. We could add some logging or something for an unhandled type of removal event.